### ISSUE TITLE
layers incorrectly rewrite extensions list

### ISSUE THREAD

**Issue by lunarpapillo** [KhronosGroup/Vulkan-ExtensionLayer#302]
This was found during SDK `vkconfig` testing.

Launch `vkconfig`. Inside `vkconfig`, launch `hwcapsviewer`, and look at the Extensions tab. Note that there is no `VK_KHR_shader_object` or `VK_KHR_synchronization2` extension reported:

![image](https://github.com/KhronosGroup/Vulkan-ExtensionLayer/assets/45772930/52f4985a-54a4-46bc-89a4-138e47cc84dc)
In `vkconfig`, create a layer configuration with `VK_LAYER_KHRONOS_synchronization2` and `VK_LAYER_KHRONOS_shader_object` forced on.  In their settings, they should both have ☑ **Force Enable** selected.  Launch `hwcapsviewer` from inside `vkconfig` again, and again check the list of extensions reported.

![image](https://github.com/KhronosGroup/Vulkan-ExtensionLayer/assets/45772930/94d8f245-c12f-4887-8ab8-a43fe6de4706)

There are two observed issues here, both with the reported list of extensions.  @christophe-lunarg suspects that one or both of the layers   `VK_LAYER_KHRONOS_synchronization2` and `VK_LAYER_KHRONOS_shader_object` fail to rewrite the reported extension list correctly.

- `VK_KHR_synchronization2` is not listed.
- `VK_EXT_vertex_attribute_divisor` was shown before the layers were activated; it disappeared after the layers were activated.

**Comment by charles-lunarg** [comment]
I went through the various options and reported what I saw in the caps viewer, which differs slightly from what is reported here. This was done on Windows using the latest SDK (1.3.261, which is in development).

The gist is that the properties & features of an extension are only present if the respective structs appear in pNext chains - which depends on the layers adding them. 

Because Sync2 doesn't define any properties, sync2 never appears in the Properties->extensions list. It does appear in the features->extensions list, as well as just the raw list of extensions.

On windows, I did not see any different in output based on whether the Force Enabled option was set in either layer.

Note that because my GPU supports sync2, they appear twice in the output.

[VulkanCapsViewer_sync2_shader_object_findings.txt](https://github.com/KhronosGroup/Vulkan-ExtensionLayer/files/12398748/VulkanCapsViewer_sync2_shader_object_findings.txt)

**Comment by christophe-lunarg** [comment]
On my machine with current test SDK:
![image](https://github.com/KhronosGroup/Vulkan-ExtensionLayer/assets/62888873/d41c5e62-1141-4289-9e10-7a1432fe116d)

**Comment by charles-lunarg** [comment]
The double reporting does look wrong - but that is probably a separate issue to the one initially reported, that the extension doesn't appear at all unless the layer is enabled through vkconfig (it may appear if the layer is enabled through other mechanisms, but I did not test that).

To respond to the two reported issues: `VK_KHR_synchronization2 is not listed in properties->extensions` because sync2 has no properties struct. It never appeared no matter what when I tested things. Don't expect sync2 to appear there. In other words, user error with VulkanCapsViewer gives misleading results.
